binary_search_tree/binary_search_tree.py

```diff
@@ -12,38 +12,164 @@ def __init__(self, value):
+            if current_tree.value == target:
+                # If it is return true
+                return True
+            # If not, go to the next tree node
+            elif current_tree.value > target:
+                next_tree = current_tree.left
+            else:
+                next_tree = current_tree.right
+        # Loop breaks when it has checked every value, if it didn't exist it returns False
+        return False
 
     # Return the maximum value found in the tree
     def get_max(self):
-        pass
+        # Declare some higher scope variables to store/keep track of values for
+        # 1: The max value
+        # 2: The current/starting tree node that we're checking and
+        # 3: The next tree node we have to check (always going to be to the right)
+        current_max = self.value
+        current_tree = self
+        next_tree = self.right
+        # While there are still tree nodes to check...
+        while next_tree is not None:
+            # Move to the next tree
+            current_tree = next_tree
+            # If that tree's value is greater than our current max value...
+            if current_tree.value > current_max:
+                # Set our current max to the new value
+                current_max = current_tree.value
+            # Continue to the next tree to the right
+            next_tree = current_tree.right
+        # Return our current max value once we hit the end of the tree
+        return current_max
 
     # Call the function `cb` on the value of each node
     # You may use a recursive or iterative approach
     def for_each(self, cb):
-        pass
+        # Call the callback function on the current/starting tree value
+        cb(self.value)
+        # Recursively iterate over each branch to the left and right,
+        # calling the callback for each tree node value
+        if self.left is not None:
+            self.left.for_each(cb)
+        if self.right is not None:
+            self.right.for_each(cb)
 
     # DAY 2 Project -----------------------
 
     # Print all the values in order from low to high
     # Hint:  Use a recursive, depth first traversal
     def in_order_print(self, node):
-        pass
+        # Check if there is a node to the left
+        if node.left:
+            # If so, call in order print recursively with that node
+            self.in_order_print(node.left)
+        # Once we hit the bottom of the left nodes, start checking the right ones
+        if node.right:
+            # Should print the value and then recur the function one to the right
+            print(node.value)
+            self.in_order_print(node.right)
+        # Once no more nodes, print the last value
+        else:
+            print(node.value)
+
 
     # Print the value of every node, starting with the given node,
     # in an iterative breadth first traversal
     def bft_print(self, node):
-        pass
+        # Create an iteration of a queue
+        queue = Queue()
```

Comment:
The magic of this is to make this a DFT, all you need to do is switch out a stack for the queue!!!!